### ISSUE TITLE
[Bug] Freedraw thickness differs between devices

### ISSUE THREAD

**Issue by josephbuchma** [excalidraw/excalidraw#5999]
See demo (iPhone 12 on the left side and Pixel 2 xl on the right side).

https://user-images.githubusercontent.com/3718145/208071160-b138f925-51d5-4723-9e40-5305923d3360.mp4

Another observation is that when I share scenes from iPhone to pixel and visa-versa (via read-only link), the thickness from the original device is preserved. For instance, a scene created on iPhone opened on both devices:

![photo_5251679370538501315_y](https://user-images.githubusercontent.com/3718145/208072484-37b94175-6147-4776-82f8-cd388b2ac7d0.jpg)

**Comment by josephbuchma** [comment]
Just checked https://perfect-freehand-example.vercel.app - same issue there, so probably issue in perfect-freehand library

**Comment by anatoliibakhtin** [comment]
It's not by chance that you have different zoom level (at the bottom left of the screen) of the canvas ? (maybe because of the different size of the screen of the devices) 

https://user-images.githubusercontent.com/71871141/208104437-a8b061a5-2ed0-4ada-b05a-c68bae2cfc04.mp4

**Comment by josephbuchma** [comment]
Zoom is 100% on both devices.

**Comment by anatoliibakhtin** [comment]
I would do the exact same test by drawing a line on the dotted line on different devices to see if the problem is that the zoom level of 100% of the canvas is relative for each device, not absolute, i.e. the problem of different ink thickness is not in the engine, but in the canvas.

p.s. I am only helping to clarify the problem, to solve it you will have to wait for programmers.

**Comment by dwelle** [comment]
@josephbuchma Testing on iPad vs Android, seems pretty much the same. If you didn't mention it reproduces on the perfect-freehand, then my immediate thought would be zoom differences as well. Sadly, on mobile it's not clear what zoom you're on. The only way to be sure you're on 100% is to clear the canvas (from the dropdown menu).

**Comment by anatoliibakhtin** [comment]
Or switch browser to pc view


https://user-images.githubusercontent.com/71871141/208305059-43dfb243-4fd1-4b19-9d00-df41045e66c5.mp4

**Comment by josephbuchma** [comment]
@dwelle I reproduced this issue in a live collab session, same devices (iPhone 12 on the  left, Pixel 2 XL on the right)

https://user-images.githubusercontent.com/3718145/208669971-26f7127f-5bbd-4faa-aeba-c1b31de7d382.mp4

**Comment by dwelle** [comment]
> @dwelle I reproduced this issue in a live collab session, same devices (iPhone 12 on the left, Pixel 2 XL on the right)
> 
>  20221219_224249.mp4

Do you happen to have that collab link so we can check the data?

**Comment by josephbuchma** [comment]
Join a live collaboration session on Excalidraw https://excalidraw.com/#room=f470f0ed2932974ac31a,K4Qc6FAxFxQkfuvtegIC0Q

**Comment by dwelle** [comment]
Thanks! It looks like an issue with pointer pressure. On Android it emits touch events with pressure, where on iOS it does not. Perhaps we should ignore pressure from touch events?

https://excalidraw.com/#json=iOrxaaIJ0WKLJ5-9iIqP8,JbQ-fTkGKVNQuFL5_ISraw

**Comment by josephbuchma** [comment]
Yeah, it emits 0.5 on Android and 0 on iOS for `event.pressure`.  Currently, it looks like the following condition determines whether the pressure is "real" or "fake": `      simulatePressure: event.pressure === 0.5,` ([link to code](https://github.com/excalidraw/excalidraw/blob/master/src/components/App.tsx#L4083))

I guess checking for `0` value there will do the trick, as long as we can safely assume that "real" pressure from things like Apple Pencil is always > 0.

**Comment by dwelle** [comment]
On Android, I'm getting (as per above link I shared) these numbers:

```
"pressures": [
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0.062745101749897,
  0.062745101749897, 0.062745101749897, 0
]
```

`simulatePressure` is an option passed down to perfect-freehand, we don't generate synthetic pressure values on our end. So Android must be emitting these numbers instead of `0.5`.

**Comment by anatoliibakhtin** [comment]
I'm sorry, I gave you an answer without going into detail (I shouldn't have to), glad to know you found the problem. You guys are awesome.

**Comment by devinat1** [comment]
I have the same issue between my android samsung tab s8+ and ipad pro.
